File: example/RotorSystem/xp.rb
#!/usr/bin/env ruby

controller = "/cys/build/control/lib/RotorSpeedController"
sim = "/cys/build/example/RotorSystem/sim/RotorSystem"
xpdir = "/cys/example/RotorSystem/control"

["rca", "rcb", "rcc"].each { |c| xp.nodes.push(controller, ["#{xpdir}/#{c}.yaml"]) }
xp.nodes.push(sim, [])

File: example/RotorSystem/xp.cxx
#include "xp.hxx"

using std::string;
using std::vector;

struct RotorXP : public XP
{
  void build() override
  {
    string 
      controller = "/cys/build/control/lib/RotorSpeedController",
      sim = "/cys/build/example/RotorSystem/sim/RotorSystem",
      xpdir = "/cys/example/RotorSystem/control",

    vector<string> controllers = {"rca", "rcb", "rcc"};

    for(auto c : controllers) {
      nodes.push_back({c, {controller, xpidr+"/"+c+".yaml"}});
    }
    nodes.push_back({ {"sim", {sim}} })
  }
};

int main()
{
  RotorXP *xp = new RotorXP();
  return xp.main();
}
